testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -643,11 +647,12 @@ void testCastExactNumericLimits(CastType castType, SqlOperatorFixture f) {
       f.checkCastToScalarOkay("'" + numeric.minNumericString + "'",
           type, numeric.minNumericString, castType);
 
-      if (Bug.CALCITE_2539_FIXED) {
+      if (numeric != Numeric.DECIMAL5_2) {
+        // The above condition is for bug CALCITE-6078
         f.checkCastFails("'" + numeric.maxOverflowNumericString + "'",
-            type, OUT_OF_RANGE_MESSAGE, true, castType);
+            type, WRONG_FORMAT_MESSAGE, true, castType);
```

Comment:
From the variable name, it seems that `OUT_OF_RANGE_MESSAGE` was the correct error we expected (similarly to the checks in the `if` statement starting at line 635)?

Reply:
Actually this variant of the test invokes something like `Integer.parseInt(string)`. So this is in fact a cast from a string to integer -- notice the single quotes added around the literal -- which gives a different error than a cast from a larger integer to an integer. The former produces a WRONG_FORMAT, the latter gives OUT_OF_RANGE. We could in principle trap the exception and rewrite the message, but it *is* a different exception in Java in the simplification code.

Reply:
The integer-to-integer cast is tested a few lines above.

Reply:
You are right, I missed the extra quote, sorry about that!